frontend/ui/widget/inputdialog.lua

```diff
@@ -456,6 +456,10 @@ function InputDialog:init()
     if self.fullscreen and not self.keyboard_visible then
         self:lockKeyboard(true)
     end
+
+    if Device:hasKeyboard() and G_reader_settings:isFalse("virtual_keyboard_enabled") then
+        self.keyboard_visible = false
```

Comment:
It's too late. Should be earlier, like
https://github.com/koreader/koreader/blob/04fe9afde7b5ab2bcf470943a4a15aff1b849620/frontend/ui/widget/inputdialog.lua#L219-L221

Reply:
Does the Terminal plugin accept these chnges?

Reply:
> Does the Terminal plugin accept these changes?

I moved it up to the top-ish of init (below that block of ifs), it works fine except for one minor thing (same applies to Style tweaks). One can never open the VK when in those widgets, in other words the `Shift` + `Home` trick does not work on those. For reference it works elsewhere just fine.